lib/platform.cpp

```diff
@@ -156,36 +157,45 @@ bool cppcheck::Platform::platform(cppcheck::Platform::PlatformType type)
     return false;
 }
 
-bool cppcheck::Platform::loadPlatformFile(const char exename[], const std::string &filename)
+bool cppcheck::Platform::loadPlatformFile(const char exename[], const std::string &filename, bool verbose)
 {
-    // open file..
-    tinyxml2::XMLDocument doc;
-    if (doc.LoadFile(filename.c_str()) != tinyxml2::XML_SUCCESS) {
-        std::vector<std::string> filenames;
-        filenames.push_back(filename + ".xml");
-        if (exename && (std::string::npos != Path::fromNativeSeparators(exename).find('/'))) {
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + filename);
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + filename);
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + filename);
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + filename + ".xml");
-        }
+    // TODO: only append .xml if missing
+    // TODO: use native separators
+    std::vector<std::string> filenames;
+    filenames.push_back(filename);
+    filenames.push_back(filename + ".xml");
+    filenames.push_back("platforms/" + filename);
+    filenames.push_back("platforms/" + filename + ".xml");
+    if (exename && (std::string::npos != Path::fromNativeSeparators(exename).find('/'))) {
+        filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + filename);
+        filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + filename);
+        filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + filename + ".xml");
+    }
 #ifdef FILESDIR
-        std::string filesdir = FILESDIR;
-        if (!filesdir.empty() && filesdir[filesdir.size()-1] != '/')
-            filesdir += '/';
-        filenames.push_back(filesdir + ("platforms/" + filename));
-        filenames.push_back(filesdir + ("platforms/" + filename + ".xml"));
+    std::string filesdir = FILESDIR;
+    if (!filesdir.empty() && filesdir[filesdir.size()-1] != '/')
+        filesdir += '/';
+    filenames.push_back(filesdir + ("platforms/" + filename));
+    filenames.push_back(filesdir + ("platforms/" + filename + ".xml"));
 #endif
-        bool success = false;
-        for (const std::string & f : filenames) {
-            if (doc.LoadFile(f.c_str()) == tinyxml2::XML_SUCCESS) {
-                success = true;
-                break;
-            }
+
+    // open file..
+    tinyxml2::XMLDocument doc;
+    bool success = false;
+    for (const std::string & f : filenames) {
+        if (verbose)
+            std::cout << "try to load platform file '" << f << "' ... ";
```

Comment:
We should not output any important information with std::cout directly. Is it just for debugging? It will not be shown properly in the GUI.

Reply:
It's verbose information. I just checked other code and that uses the `ErrorLogger::reportOut()`. Will adjust.

Reply:
Turns out that is not the function to use since it is not available at that point.

Also the rest of the code is not doing this consistently either. So I propose to apply these changes as is. With a later PR I will refactor out the explicit `std::cout` and `std::cerr` usage as a whole.